### ISSUE TITLE
Cannot use file from other source as Helm --set-file param

### ISSUE THREAD

**Issue by lucacox** [argoproj/argo-cd#13220]
Checklist:

* [x] I've searched in the docs and FAQ for my answer: https://bit.ly/argocd-faq.
* [x] I've included steps to reproduce the bug.
* [x] I've pasted the output of `argocd version`.

**Describe the bug**

In Application using multiple sources reference of one source is not substituted in helm.fileParameters entries. It works (at least it seems) for the helm.valueFiles property

**To Reproduce**

Using this Application definition:

```yaml
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: kratos
  namespace: argocd
spec:
  project: ory
  destination:
    name: ''
    namespace: auth
    server: 'https://kubernetes.default.svc'
  sources:
    - repoURL: 'https://k8s.ory.sh/helm/charts'
      targetRevision: 0.30.0
      chart: kratos
      helm:
        valueFiles:
          - $myRepo/kratos/k8s/overlay/development/kratos-values.yaml

        fileParameters:
          - name: identity
            path: $myRepo/kratos/k8s/overlay/development/identity_schemas/identity.schema.json
    - repoURL: git@my-private:repo.git
      targetRevision: development
      ref: myRepo
```
gives this error in the UI:

```
rpc error: code = Unknown desc = Manifest generation error (cached): `helm template . --name-template kratos --namespace auth --kube-version 1.25 --set-file identity=/tmp/47de275d-f0c3-419a-8382-9628f1674103/kratos/kratos/k8s/overlay/development/identity_schemas/identity.schema.json --values .kratos/k8s/overlay/development/kratos-values.yaml --api-versions acid.zalan.do/v1 --api-versions acid.zalan.do/v1/OperatorConfiguration --api-versions acid.zalan.do/v1/PostgresTeam --api-versions acid.zalan.do/v1/postgresql --api-versions admissionregistration.k8s.io/v1 --api-versions admissionregistration.k8s.io/v1/MutatingWebhookConfiguration --api-versions admissionregistration.k8s.io/v1/ValidatingWebhookConfiguration --api-versions apiextensions.k8s.io/v1 --api-versions apiextensions.k8s.io/v1/CustomResourceDefinition --api-versions apiregistration.k8s.io/v1 --api-versions apiregistration.k8s.io/v1/APIService --api-versions apps/v1 --api-versions apps/v1/ControllerRevision --api-versions apps/v1/DaemonSet --api-versions apps/v1/Deployment --api-versions apps/v1/ReplicaSet --api-versions apps/v1/StatefulSet --api-versions argoproj.io/v1alpha1 --api-versions argoproj.io/v1alpha1/AppProject --api-versions argoproj.io/v1alpha1/Application --api-versions argoproj.io/v1alpha1/ApplicationSet --api-versions authzed.com/v1alpha1 --api-versions authzed.com/v1alpha1/SpiceDBCluster --api-versions auto.gke.io/v1 --api-versions auto.gke.io/v1/AllowlistedV2Workload --api-versions auto.gke.io/v1/AllowlistedWorkload --api-versions auto.gke.io/v1alpha1 --api-versions auto.gke.io/v1alpha1/AllowlistedWorkload --api-versions autoscaling/v1 --api-versions autoscaling/v1/HorizontalPodAutoscaler --api-versions autoscaling/v2 --api-versions autoscaling/v2/HorizontalPodAutoscaler --api-versions autoscaling/v2beta2 --api-versions autoscaling/v2beta2/HorizontalPodAutoscaler --api-versions batch/v1 --api-versions batch/v1/CronJob --api-versions batch/v1/Job --api-versions certificates.k8s.io/v1 --api-versions certificates.k8s.io/v1/CertificateSigningRequest --api-versions cloud.google.com/v1 --api-versions cloud.google.com/v1/BackendConfig --api-versions cloud.google.com/v1beta1 --api-versions cloud.google.com/v1beta1/BackendConfig --api-versions coordination.k8s.io/v1 --api-versions coordination.k8s.io/v1/Lease --api-versions core.strimzi.io/v1beta2 --api-versions core.strimzi.io/v1beta2/StrimziPodSet --api-versions discovery.k8s.io/v1 --api-versions discovery.k8s.io/v1/EndpointSlice --api-versions events.k8s.io/v1 --api-versions events.k8s.io/v1/Event --api-versions flowcontrol.apiserver.k8s.io/v1beta1 --api-versions flowcontrol.apiserver.k8s.io/v1beta1/FlowSchema --api-versions flowcontrol.apiserver.k8s.io/v1beta1/PriorityLevelConfiguration --api-versions flowcontrol.apiserver.k8s.io/v1beta2 --api-versions flowcontrol.apiserver.k8s.io/v1beta2/FlowSchema --api-versions flowcontrol.apiserver.k8s.io/v1beta2/PriorityLevelConfiguration --api-versions getambassador.io/v1beta1 --api-versions getambassador.io/v1beta1/RateLimit --api-versions getambassador.io/v1beta2 --api-versions getambassador.io/v1beta2/Filter --api-versions getambassador.io/v1beta2/FilterPolicy --api-versions getambassador.io/v1beta2/RateLimit --api-versions getambassador.io/v2 --api-versions getambassador.io/v2/AuthService --api-versions getambassador.io/v2/ConsulResolver --api-versions getambassador.io/v2/DevPortal --api-versions getambassador.io/v2/Filter --api-versions getambassador.io/v2/FilterPolicy --api-versions getambassador.io/v2/Host --api-versions getambassador.io/v2/KubernetesEndpointResolver --api-versions getambassador.io/v2/KubernetesServiceResolver --api-versions getambassador.io/v2/LogService --api-versions getambassador.io/v2/Mapping --api-versions getambassador.io/v2/Module --api-versions getambassador.io/v2/RateLimit --api-versions getambassador.io/v2/RateLimitService --api-versions getambassador.io/v2/TCPMapping --api-versions getambassador.io/v2/TLSContext --api-versions getambassador.io/v2/TracingService --api-versions getambassador.io/v3alpha1 --api-versions getambassador.io/v3alpha1/AuthService --api-versions getambassador.io/v3alpha1/ConsulResolver --api-versions getambassador.io/v3alpha1/DevPortal --api-versions getambassador.io/v3alpha1/Filter --api-versions getambassador.io/v3alpha1/FilterPolicy --api-versions getambassador.io/v3alpha1/Host --api-versions getambassador.io/v3alpha1/KubernetesEndpointResolver --api-versions getambassador.io/v3alpha1/KubernetesServiceResolver --api-versions getambassador.io/v3alpha1/Listener --api-versions getambassador.io/v3alpha1/LogService --api-versions getambassador.io/v3alpha1/Mapping --api-versions getambassador.io/v3alpha1/Module --api-versions getambassador.io/v3alpha1/RateLimit --api-versions getambassador.io/v3alpha1/RateLimitService --api-versions getambassador.io/v3alpha1/TCPMapping --api-versions getambassador.io/v3alpha1/TLSContext --api-versions getambassador.io/v3alpha1/TracingService --api-versions hub.gke.io/v1 --api-versions hub.gke.io/v1/Membership --api-versions internal.autoscaling.gke.io/v1alpha1 --api-versions internal.autoscaling.gke.io/v1alpha1/CapacityRequest --api-versions kafka.strimzi.io/v1alpha1 --api-versions kafka.strimzi.io/v1alpha1/KafkaTopic --api-versions kafka.strimzi.io/v1alpha1/KafkaUser --api-versions kafka.strimzi.io/v1beta1 --api-versions kafka.strimzi.io/v1beta1/KafkaTopic --api-versions kafka.strimzi.io/v1beta1/KafkaUser --api-versions kafka.strimzi.io/v1beta2 --api-versions kafka.strimzi.io/v1beta2/Kafka --api-versions kafka.strimzi.io/v1beta2/KafkaBridge --api-versions kafka.strimzi.io/v1beta2/KafkaConnect --api-versions kafka.strimzi.io/v1beta2/KafkaConnector --api-versions kafka.strimzi.io/v1beta2/KafkaMirrorMaker --api-versions kafka.strimzi.io/v1beta2/KafkaMirrorMaker2 --api-versions kafka.strimzi.io/v1beta2/KafkaRebalance --api-versions kafka.strimzi.io/v1beta2/KafkaTopic --api-versions kafka.strimzi.io/v1beta2/KafkaUser --api-versions migration.k8s.io/v1alpha1 --api-versions migration.k8s.io/v1alpha1/StorageState --api-versions migration.k8s.io/v1alpha1/StorageVersionMigration --api-versions monitoring.googleapis.com/v1 --api-versions monitoring.googleapis.com/v1/ClusterPodMonitoring --api-versions monitoring.googleapis.com/v1/ClusterRules --api-versions monitoring.googleapis.com/v1/GlobalRules --api-versions monitoring.googleapis.com/v1/OperatorConfig --api-versions monitoring.googleapis.com/v1/PodMonitoring --api-versions monitoring.googleapis.com/v1/Rules --api-versions monitoring.googleapis.com/v1alpha1 --api-versions monitoring.googleapis.com/v1alpha1/ClusterPodMonitoring --api-versions monitoring.googleapis.com/v1alpha1/ClusterRules --api-versions monitoring.googleapis.com/v1alpha1/GlobalRules --api-versions monitoring.googleapis.com/v1alpha1/OperatorConfig --api-versions monitoring.googleapis.com/v1alpha1/PodMonitoring --api-versions monitoring.googleapis.com/v1alpha1/Rules --api-versions networking.gke.io/v1 --api-versions networking.gke.io/v1/ManagedCertificate --api-versions networking.gke.io/v1/ServiceAttachment --api-versions networking.gke.io/v1beta1 --api-versions networking.gke.io/v1beta1/FrontendConfig --api-versions networking.gke.io/v1beta1/ManagedCertificate --api-versions networking.gke.io/v1beta1/ServiceAttachment --api-versions networking.gke.io/v1beta1/ServiceNetworkEndpointGroup --api-versions networking.gke.io/v1beta2 --api-versions networking.gke.io/v1beta2/ManagedCertificate --api-versions networking.k8s.io/v1 --api-versions networking.k8s.io/v1/Ingress --api-versions networking.k8s.io/v1/IngressClass --api-versions networking.k8s.io/v1/NetworkPolicy --api-versions node.k8s.io/v1 --api-versions node.k8s.io/v1/RuntimeClass --api-versions nodemanagement.gke.io/v1alpha1 --api-versions nodemanagement.gke.io/v1alpha1/UpdateInfo --api-versions oathkeeper.ory.sh/v1alpha1 --api-versions oathkeeper.ory.sh/v1alpha1/Rule --api-versions policy/v1 --api-versions policy/v1/PodDisruptionBudget --api-versions rbac.authorization.k8s.io/v1 --api-versions rbac.authorization.k8s.io/v1/ClusterRole --api-versions rbac.authorization.k8s.io/v1/ClusterRoleBinding --api-versions rbac.authorization.k8s.io/v1/Role --api-versions rbac.authorization.k8s.io/v1/RoleBinding --api-versions redis.redis.opstreelabs.in/v1beta1 --api-versions redis.redis.opstreelabs.in/v1beta1/Redis --api-versions redis.redis.opstreelabs.in/v1beta1/RedisCluster --api-versions scheduling.k8s.io/v1 --api-versions scheduling.k8s.io/v1/PriorityClass --api-versions snapshot.storage.k8s.io/v1 --api-versions snapshot.storage.k8s.io/v1/VolumeSnapshot --api-versions snapshot.storage.k8s.io/v1/VolumeSnapshotClass --api-versions snapshot.storage.k8s.io/v1/VolumeSnapshotContent --api-versions snapshot.storage.k8s.io/v1beta1 --api-versions snapshot.storage.k8s.io/v1beta1/VolumeSnapshot --api-versions snapshot.storage.k8s.io/v1beta1/VolumeSnapshotClass --api-versions snapshot.storage.k8s.io/v1beta1/VolumeSnapshotContent --api-versions storage.k8s.io/v1 --api-versions storage.k8s.io/v1/CSIDriver --api-versions storage.k8s.io/v1/CSINode --api-versions storage.k8s.io/v1/CSIStorageCapacity --api-versions storage.k8s.io/v1/StorageClass --api-versions storage.k8s.io/v1/VolumeAttachment --api-versions storage.k8s.io/v1beta1 --api-versions storage.k8s.io/v1beta1/CSIStorageCapacity --api-versions v1 --api-versions v1/ConfigMap --api-versions v1/Endpoints --api-versions v1/Event --api-versions v1/LimitRange --api-versions v1/Namespace --api-versions v1/Node --api-versions v1/PersistentVolume --api-versions v1/PersistentVolumeClaim --api-versions v1/Pod --api-versions v1/PodTemplate --api-versions v1/ReplicationController --api-versions v1/ResourceQuota --api-versions v1/Secret --api-versions v1/Service --api-versions v1/ServiceAccount --api-versions warden.gke.io/v1 --api-versions warden.gke.io/v1/Audit --include-crds` failed exit status 1: Error: failed parsing --set-file data: open /tmp/47de275d-f0c3-419a-8382-9628f1674103/kratos/kratos/k8s/overlay/development/identity_schemas/identity.schema.json: no such file or directory
14 minutes ago (Thu Apr 13 2023 17:01:24 GMT+0200)
```

Notice this:

```
--set-file identity=/tmp/47de275d-f0c3-419a-8382-9628f1674103/kratos/kratos/k8s/overlay/development/identity_schemas/identity.schema.json 
--values .kratos/k8s/overlay/development/kratos-values.yaml
```

in `--values` param the `$myRepo` is substituted with a `.` and in `--set-file` with `/tmp/47de275d-f0c3-419a-8382-9628f1674103/kratos`.

**Expected behavior**

The files in the gite repo should be used to set the variables in Helm

**Screenshots**

<img width="703" alt="image" src="https://user-images.githubusercontent.com/1597231/231807535-18c1e7f5-875c-4e74-9d93-bf1f32304bee.png">

**Version**

```shell
argocd: v2.6.7+5bcd846
  BuildDate: 2023-03-23T14:57:27Z
  GitCommit: 5bcd846fa16e4b19d8f477de7da50ec0aef320e5
  GitTreeState: clean
  GoVersion: go1.18.10
  Compiler: gc
  Platform: linux/amd64
```

**Comment by ishitasequeira** [comment]
@lucacox, multiple sources is not yet supported for `helm.fileParameters` field. It is only supported for `helm.valueFiles` field as of now.

Were you able to see the error if you reference value file from other source only, that is, only reference `$myRepo` in `helm.valueFiles`?

**Comment by mnikser** [comment]
Supporting helm.fileParameters field under the multiple sources is very useful. I also have use case for and I noticed it is not supported. Is there a place where we can ask for features to be added? Or is this on your road map to be added?

**Comment by crenshaw-dev** [comment]
This PR would solve: https://github.com/argoproj/argo-cd/pull/12508

**Comment by m-adawi** [comment]
I'm having the same issue. I'm using a centralized [helm chart](https://bitnami.com/stack/aspnet-core)  to deploy multiple ASP.NET services instead of replicating the same helm chart in many repos. I'm using the helm value extraDeploy this way:

```yaml
extraDeploy: 
  - |
    apiVersion: v1
    kind: ConfigMap
    metadata:
      name: {{ include "aspnet-core.fullname" . }}-appsettings
      labels: {{- include "common.labels.standard" . | nindent 6 }}
        {{- if .Values.commonLabels }}
        {{- include "common.tplvalues.render" ( dict "value" .Values.commonLabels "context" $ ) | nindent 6 }}
        {{- end }}
      {{- if .Values.commonAnnotations }}
      annotations: {{- include "common.tplvalues.render" ( dict "value" .Values.commonAnnotations "context" $ ) | nindent 6 }}
      {{- end }}
    data:
      appsettings.json: |-
        {{- .Values.appsettings | nindent 4 }}
```

So I need to set the helm value appsettings to the appsettings.json file content

**Comment by jaydipdave** [comment]
This is a high priority issue!! My prod app has stopped working, it was working before in the old version.

**Comment by illnr** [comment]
I found a solution for my use case here: https://github.com/argoproj/argo-cd/issues/3365#issuecomment-617414414 
You can use an umbrella chart, like [this one](https://github.com/argoproj/argocd-example-apps/tree/master/helm-dependency). You have your own chart in git and can have the `valueFiles` and `fileParameters` in the same git repo, because the actual chart is a dependency.  
So the only overhead is to create a helm umbrella chart everywhere you want to use `fileParameters`.

**Comment by cardoe** [comment]
I created an implementation of this in #17941, hoping someone that's interested in this feature can help me get it over the finish line.

**Comment by cardoe** [comment]
Thanks to reviews from @morremeyer I think the PR is in good shape. I've personally cherry picked it into the v2.11.0 release and built it locally and I'm using that in our dev/test environment with success.

**Comment by spy16** [comment]
Is this feature merged or any plan on when it will be merged? We also need this feature

**Comment by cardoe** [comment]
This is fixed once #17941 is merged.

**Comment by patpicos** [comment]
Is this limitation in the documentation still in effect or that has been resolved? 
<img width="479" alt="image" src="https://github.com/user-attachments/assets/aca3ad36-06cf-4ae0-940e-4540e3d6e2b5">

**Comment by morremeyer** [comment]
@patpicos See https://github.com/argoproj/argo-cd/pull/17941#issuecomment-2270726002, it will be released with v1.13.0
